crates/core/src/transport/symmetric_message.rs

```diff
@@ -11,50 +11,57 @@ use super::{packet_data::MAX_PACKET_SIZE, PacketData};
 pub(super) struct SymmetricMessage {
     // todo: make sure we handle wrapping around the u16 properly
     pub message_id: u16,
+    // todo: profile what is better here on average in the future
+    // (vec, fixed array size of what given length etc.
     #[serde_as(as = "Option<[_; 50]>")]
```

Comment:
Why a fixed length? 100 bytes of every 1500 byte message seems like a lot given that most of the time there will probably be far fewer confirmations required.

Reply:
this stuff is better profilling... allocating on the heap may be more expensive that just reserving 100 bytes in most modern processors that is just 12 word size which fit in cache; is nothing really.

but we will see

Reply:
when serialized this could end up just taking as many u16 as needed and no more, so it won't be 100 bytes, if there are only 2 ids it could be just 4 bytes

Reply:
I see, so this is just about the in-memory representation of the packet, not the serialized packet that's sent over the network? I'm not sure I follow but it sounds like you know what you're doing ;)